support.js

```diff
@@ -72,13 +72,15 @@ const registerHooks = () => {
         return
       }
 
-      if (
-        Cypress._.find(windowCoverageObjects, {
-          coverage: applicationSourceCoverage
-        })
-      ) {
+      const existingCoverage = Cypress._.find(windowCoverageObjects, {
+        coverage: applicationSourceCoverage
+      })
+      if (existingCoverage) {
         // this application code coverage object is already known
         // which can happen when combining `window:load` and `before` callbacks
+        // it can also happen when the user leaves and returns to the application under test
+        // in which case we need to use new applicationSourceCoverage, because the old will not be updated anymore.
+        existingCoverage.coverage = applicationSourceCoverage
```

Comment:
Does this cause any of the prior code coverage to be lost or does `applicationSourceCoverage` also contain the prior coverage?

Reply:
It shouldn't. There is an equality comparison a few lines above, so only coverage objects with equivalent content should be replaced.

Reply:
The coverage objects being replaced is not updated anymore and the new one replacing it has the some content.